hack/run-perf-profile-creator.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+readonly CURRENT_SCRIPT=$(basename "$0")
+readonly DATA_DIR="./must-gather-content"
+readonly CMD="podman run --entrypoint performance-profile-creator"
+readonly MUST_GATHER_VOL="/must-gather"
+readonly MUST_GATHER_CMD="oc adm must-gather --request-timeout=10m"
+readonly IMG_EXISTS_CMD="podman image exists"
+readonly IMG_PULL_CMD="podman image pull"
+
+PAO_IMG="quay.io/openshift-kni/performance-addon-operator:4.8-snapshot"
+MG_IMG="quay.io/openshift-kni/performance-addon-operator-must-gather:4.8-snapshot"
+MG_TARBALL=""
+ERASE_DATA_DIR="n"
+
+usage() {
+  print "Wrapper usage:"
+  print "  ${CURRENT_SCRIPT} [-h] [-p image] [-m image] [-t path] [-e]  -- [performance-profile-creator flags]"
+  print ""
+  print "Options:"
+  print "   -h                 help for ${CURRENT_SCRIPT}"
+  print "   -p                 Performance Addon Operator image"
+  print "   -m                 Must-gather image"
+  print "   -t                 Must-gather tarball"
+  print "   -e                 erase must-gather cache"
+
+  ${IMG_EXISTS_CMD} "${PAO_IMG}" && ${CMD} "${PAO_IMG}" -h
+}
+
+exit_error() {
+  print "error: $*"
+  usage
+  exit 1
+}
+
+print() {
+  echo  "$*" >&2
+}
+
+check_requirements() {
+  [ -d "${DATA_DIR}" ] || mkdir "${DATA_DIR}"
+
+  ${IMG_EXISTS_CMD} "${PAO_IMG}" || ${IMG_PULL_CMD} "${PAO_IMG}" || \
+      exit_error "Performance Addon Operator image not found"
+
+  [ -z "${MG_TARBALL}" ] || [ -f "${MG_TARBALL}" ] || \
+      exit_error "Must-gather tarball file not found"
+
+  [ -n "${MG_TARBALL}" ] || ${IMG_EXISTS_CMD} "${MG_IMG}" || ${IMG_PULL_CMD} "${MG_IMG}" || \
+      exit_error "Must-gather image not found and must-gather tarball not provided"
+
+  if [ -f "${MG_TARBALL}" ]; then
+      rm -rf "${DATA_DIR:?}"/*
+      tar -zxf "${MG_TARBALL}" --strip-components=1 --directory "${DATA_DIR}" || \
+          exit_error "Can't decompress must-gather tarball"
+ else
+      [ ${ERASE_DATA_DIR} = "y" ] && rm -rf "${DATA_DIR:?}"/*
```

Comment:
the trailing slash makes me worried. Can ever DATA_DIR be empty?

Reply:
As of now, yes, since we create the directory is is not there.
Yes, we need it empty so we can differentiate the case when we run the must-gather tool and the case we already run it.